oeps/architectural-decisions/oep-0058-arch-translations-management.rst

```diff
@@ -0,0 +1,263 @@
+OEP-58: Translations Management
+###############################
+
+.. list-table::
+   :widths: 25 75
+
+   * - OEP
+     - :doc:`OEP-58 <oep-0058-arch-translations-management>`
+   * - Title
+     - Translations Management
+   * - Last Modified
+     - 2022-11-07
+   * - Authors
+     - 
+       * Carlos Muniz <cmuniz@tcril.org>
+       * Feanil Patel <feanil@tcril.org>
+       * Sarina Canelake <sarina@tcril.org>
+   * - Arbiter
+     - Ned Batchelder <ned@edx.org>
+   * - Status
+     - Under Review
+   * - Type
+     - Architecture Decision
+   * - Created
+     - 2022-08-08
+   * - Review Period
+     - 2022-11-07 - 2022-11-21
+..    * - Resolution
+..      - 
+
+.. contents::
+  :local:
+  :depth: 1
+
+Context
+*******
+
+The current method of managing and organizing translations files is overly complicated
+and unavailable to the majority of the Open edX community. For example: the
+edx-transifex-bot performs the automated upload of English translation source files to
+Transifex and download of translation files to GitHub. It currently runs on legacy
+infrastructure originally provided by a community member and is difficult to track why
+some PRs are merged by the bot, and some are not, and where the bot is creating and
+merging PRs. Most recently, it was discovered that the translations were not uploading
+properly but it has been impossible for most members of the Open edX community to debug
+exactly why. In the week before the Nutmeg release, this was a significant pain point.
+
+Decision
+********
+
+To alleviate these issues, we will switch from using the edx-transifex-bot to the
+`Transifex GitHub App`_, a stable app provided by Transifex. Benefits of this change
+include being easier to maintain and solving a lot of the pain points detailed below. As
+part of this proposal, we suggest moving translations into their own repository, to make
+using the `Transifex GitHub App`_ more streamlined and straightforward, and in order to
+make organizing and using the up to date translations simpler.
+
+Current State
+*************
+
+* edx-transifex-bot is a potential security issue: The edx-transifex-bot requires admin
+  rights on Transifex in order to function. Admin rights give access to private/sensitive
+  information as well as the ability to permanently delete translation and configuration
+  files. At some point, the login to the edx-transifex-bot user was lost, and without
+  access to infrastructure that the bot uses to function, this edx-transifex-bot is a
+  security issue most of the Open edX community cannot control or debug.
+* edx-transifex-bot is a black box for most of the community: the code for the
+  edx-transifex-bot is in the `ecommerce-scripts`_ repository but it is impossible for
+  most of the community to observe the work it is doing, or whether it is doing it
+  correctly. In addition, there is no documentation for these important scripts.
```

Comment:
Where are the errors of the Transifex GitHub App shown? I understand that many of the errors will be caught on extraction by the GitHub workflow (and thus will be visible via PR), but I wonder if there are errors that will only be caught at the Transifex GitHub App level.

Assuming such errors exist (they may not):

1. What happens if two projects happen to use the same translation key for a string? (Of course, we have a naming convention to prevent that, but say a copy and paste error?)
2. And will an error caused by one project block translations by all other projects?

Reply:
I think this is a valid concern. Transifex has a very robust notification system, and errors occurring there will notify the Transifex Admins.

For your thought experiments, Transifex projects are treated exclusively. What happens in one, does not affect the others, similar to how GitHub treats repositories. So an error in any Transifex project other than openedx-translations will not affect the translations in Transifex project. 

As a side note, translation files were kept in their original file structures to combat the non-unique file name errors we have experienced before, as the entire path is used as the unique translation slug. So as long as files are kept in separate directory structures within the openedx-translations repository, this will not be a problem.

Reply:
Ok, great, thanks for the clarification! My sense is that we should be good as long as there is a way for maintainers to know that changes they made caused errors at the Transifex level. It sounds like the Translation WG or Transifex Admins are taking up this role and will be responsible for notifying the relevant maintainer?

I should note that I was being a little confusing in my parent comment because I used "project" to refer to an Open edX Project (a repository) instead of a Transifex Project. I was just trying to make sure an error caused by one repository will not block translations for all other repositories. (I suspect that may have been one of the reasons the original Jenkins jobs that pushed/pulled translations were done on a per-respository basis.)